src/transport/AdminPairingTable.h

```diff
@@ -46,6 +49,16 @@ struct AccessControlList
     uint32_t placeholder;
 };
 
+class DLL_EXPORT AdminPairingTableDelegate
+{
+public:
+    virtual ~AdminPairingTableDelegate() {}
+
+    virtual void OnAdminDeletedFromStorage(AdminId adminId);
+    virtual void OnAdminRetrievedFromStorage(AdminId adminId, FabricId fabricId, NodeId nodeId);
+    virtual void OnAdminPersistedToStorage(AdminId adminId, FabricId fabricId, NodeId nodeId);
```

Comment:
The semantics of these callbacks need to be clearly documented.  Especially given the weirdness around async delete, so that `OnAdminDeletedFromStorage` gets called before the delete has happened (always?  sometimes?).  But need to generally document how these callbacks order with the underlying KVS mutations.

Reply:
This will all change. I do not want two sources of truths. Can document but I am planning a rewrite as the layering is not right and we shouldn't be storing things in both the admin pairing table, persisted storage and the attribute store. The access to the admin pairing table global should not live in server.cpp either. There should be one access point to modify/add/delete admins that then modifies the persisted storage...

Reply:
OK.  If this is temporary code, please document, link to issue that tracks fixing it.